### ISSUE TITLE
Binary matrix as input

### ISSUE THREAD

**Issue by remiadon** [scikit-mine/scikit-mine#8]
#### Workflow
For many people in the machine learning community representing transactional datasets as we do in skmine is not something usual
Moreover, a common workflow we will encounter at some point consists in using the output of our Transformers as input to sklearn.

In pattern mining researchers are already familiar with matrix representations of transactional databases. Quoting Vreeken and Al. from [SLIM](https://eda.mmci.uni-saarland.de/pubs/2012/slim_directly_mining_descriptive_patterns-smets,vreeken.pdf), Section 2.1

> Note that any binary or categorical dataset can be trivially converted into a transaction database

#### Proposed solution
To bridge this gap we should show some example working with a transactional dataset
We should be able to 
 - transform a standard dataset (eg. chess) into a binary matrix
- inverse transforming should not be very useful, considering the use cases

The most straight-forward solution seems to be using the [sklearn.preprocessing.MultilabelBinarizer](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MultiLabelBinarizer.html#sklearn.preprocessing.MultiLabelBinarizer), in this way

```python
>>> from sklearn.preprocessing import MultiLabelBinarizer
>>> from skmine.datasets.fimi import fetch_chess
>>> D = fetch_chess()
>>> mlb = MultiLabelBinarizer()
>>> X = mlb.fit_transform(D)
>>> X
array([[1, 0, 1, ..., 0, 0, 1],
       [1, 0, 0, ..., 0, 0, 1],
       [1, 0, 0, ..., 0, 0, 1],
       ...,
       [0, 0, 1, ..., 0, 0, 1],
       [0, 0, 1, ..., 0, 1, 1],
       [0, 0, 1, ..., 0, 1, 1]])
```

#### Describe alternatives you've considered, if relevant
If scikit-learn does not fit out purpose we can still implement our own transformer.
But the preferred solution would be to make a PR to scikit-learn

#### Additional context
Note that the `MutliLabelBinarizer` is only suitable to mere itemsets, it does not work out of the box on eg. sequential itemsets.
Even if it can be twicked for this purpose, eg:
```python
>>> s = pd.Series([[2, 3, 2], [0, 2]])  # 2 is present at two different positions in the first transaction
>>> mlb = MultiLabelBinarizer()
>>> mlb.fit_transform(s.map(enumerate))  # note that inverse_transform will fail to reconstruct the original input
array([[0, 1, 0, 1, 1, 1],
       [1, 0, 1, 0, 0, 0]])
>>> mbl.classes_
array([(0, 0), (0, 2), (1, 2), (1, 3), (2, 4), (3, 2)], dtype=object)
```

**Comment by remiadon** [comment]
UPDATE

Considering only basic transactional datasets, MultiLabelBinarizer gives access to a `.classes_` attributes. The problem is that it is an attribute of the object, **it does not transit within data** --> sklearn MultiLabelBinarizer produces np.array or scipy.sparse matrics, but they are not labels

In the context of Pattern mining we absolutely need the labels : theses are our beloved symbols$

```python
>>> from skmine.preprocessing import MulitLabelBinarizer
>>> mb = MultiLabelBinarizer(sparse_output=True)
>>> D = pd.Series([  # SLIM takes a pd.Series as input  
>>>    ['bananas', 'milk'],  
>>>     ['milk', 'bananas', 'cookies'],  
>>>   ['cookies', 'butter', 'tea'],  
>>>    ['tea'],   
>>>    ['milk', 'bananas', 'tea'],  
>>>  ])
>>> tab_D = mb.fit_transform(D)  # scipy sparse matrix
>>> tab_D[:2].todense()
matrix([[1, 0, 0, 1, 0],
        [1, 0, 1, 1, 0]])
>>> mb.classes_
array(['bananas', 'butter', 'cookies', 'milk', 'tea'], dtype=object)
```


My solution:
- reimplement skearn MultiLabelBinarizer to make its `transform` function return a pandas.DataFrame with some explicit columns (our symbols)
- propose this in sklearn as a feature request. If accepted, we will use it as implemented in sklearn to make sure we don't reimplement things twice

**Comment by remiadon** [comment]
see 
https://github.com/scikit-mine/scikit-mine/pull/35/commits/e4b467b7069a37ee47cff14b282f0e4f85b2b076
